art_internal.hpp

```diff
@@ -167,10 +204,16 @@ class [[nodiscard]] basic_node_ptr {
   explicit basic_node_ptr(std::nullptr_t) noexcept
       : tagged_ptr{reinterpret_cast<std::uintptr_t>(nullptr)} {}
 
+  // construct a node pointer given a raw pointer and a node type.
   basic_node_ptr(header_type *ptr UNODB_DETAIL_LIFETIMEBOUND,
                  unodb::node_type type) noexcept
       : tagged_ptr{tag_ptr(ptr, type)} {}
 
+  // constructor casts away [const] for use when the node_ptr will be [const].
+  basic_node_ptr(const header_type *ptr UNODB_DETAIL_LIFETIMEBOUND,
+                 unodb::node_type type) noexcept
+      : tagged_ptr{tag_ptr(const_cast<header_type *>(ptr), type)} {}
```

Comment:
Can we drop `const_cast` here and add `const` to `*ptr_` arg for `tag_ptr` method?